storage/series/series.go

```diff
@@ -440,6 +442,31 @@ func (s *dbSeries) mergeBlock(
 	return nil
 }
 
+func (s *dbSeries) Update(blk block.DatabaseBlock) error {
+	s.Lock()
+	defer s.Unlock()
+
+	// ensure series is bootstraped before allowing any updates
+	if s.bs != bootstrapped {
+		return errSeriesNotBootstrapped
+	}
+
+	// we only accept updates to non-buffered data
+	// ensure buffer is past provided time
+	min, _ := s.buffer.MinMax()
+	if !blk.StartTime().Before(min) {
```

Comment:
The `MinMax` looks to be set independent of the datapoints - https://github.com/m3db/m3db/blob/prateek/shard-repairer/storage/series/buffer.go#L210-L220 and https://github.com/m3db/m3db/blob/prateek/shard-repairer/storage/series/buffer.go#L345